src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/PackageReferenceProject.cs

```diff
@@ -159,12 +158,17 @@ public virtual async Task<ProjectPackages> GetInstalledAndTransitivePackagesAsyn
 
             CounterfactualLoggers.TransitiveDependencies.EmitIfNeeded(); // Emit only one event per VS session
             IEnumerable<TransitivePackageReference> transitivePackagesWithOrigins;
-            if (await ExperimentUtility.IsTransitiveOriginExpEnabled.GetValueAsync(token))
+            if (await ExperimentUtility.IsTransitiveOriginExpEnabled.GetValueAsync(token) && useTransitiveOrigins)
             {
                 // Compute Transitive Origins
                 Dictionary<string, TransitiveEntry> transitiveOrigins;
                 if (IsInstalledAndTransitiveComputationNeeded || TransitiveOriginsCache == null)
                 {
+                    if (targetsList == null) // special case
+                    {
+                        targetsList = (await GetTargetsListAsync(assetsPath, token))?.ToList();
+                    }
```

Comment:
Added a comment. we need to load assets file when Installed and Transitive lists are populated but no populated Transitive Origins cache populated. This can happen when you call GetInstalledAndTransitivePackagesAsync() with `includeTransitiveOrigins` set to false.